Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -50,21 +52,22 @@ import Foundation
         case .tokenizationKey:
             do {
                 self.authorization =  try TokenizationKey(authorization)
-                http = BTHTTP(authorization: self.authorization)
             } catch {
                 return nil
             }
         case .clientToken:
             do {
                 let clientToken = try BTClientToken(clientToken: authorization)
                 self.authorization = clientToken
-
-                http = BTHTTP(authorization: self.authorization)
             } catch {
                 return nil
             }
         }
         
+        let btHttp = BTHTTP(authorization: self.authorization)
+        http = btHttp
```

Comment:
```suggestion
        http = BTHTTP(authorization: self.authorization)
```

Reply:
yeah, i dont like it neeither, I've created this:

`let btHttp = BTHTTP(authorization: self.authorization)
`
because `BTAPIClient.http` is optional and` ConfigurationLoader.init` needs a `non-optional` instance of BTHTTP.

> configurationLoader = ConfigurationLoader(http: btHttp)

Reply:
This ensures that I have the same instance for both `BTAPIClient` and `ConfigurationLoader`, and it avoids having to use "!" when instantiating the `ConfigurationLoader`.

```
let btHttp = BTHTTP(authorization: self.authorization)
http = btHttp
configurationLoader = ConfigurationLoader(http: btHttp)
```

Reply:
Ah, that makes sense!